download-data.py

```diff
@@ -424,6 +461,37 @@ def main() -> None:
             file=sys.stderr,
         )
 
+
+    ### Update symlink if requested, without downloading data, if the target directory exists
+    if args.update_symlink:
+
+        # find all non-hidden directories in data_dir, representing the local releases
+        local_releases = [x.stem for x in args.data_dir.iterdir() if re.match("^[^.]", x.name) and x.is_dir()]
+
+        if args.test_data:
+            target = "test"
+        else:
+            target = args.release
+
+        if not target in local_releases:
+            if args.dryrun:
+                print(f"\nThe '{target}' data release is not locally available. The symlink would not be updated.")
+            else:
+                print(f"\nThe '{target}' data release is not locally available. Could not update symlink.")
+            sys.exit(1)
+        else:
+            if target == "current":
+                # find the most recent local release
+                dated_releases = [x for x in local_releases if re.match("\d{4}-\d{2}-\d{2}", x)]
```

Comment:
move the `local_releases` here, because we don't need to look at all except when "current" is used. You can probably eliminate the variable completely by combining the two comprehensions:
```suggestion
                dated_releases = [x.name for x in args.data_dir.iterdir() if x.is_dir and re.match("\d{4}-\d{2}-\d{2}$", x.name)]
```